openedx/core/djangoapps/content_libraries/keys.py

```diff
@@ -0,0 +1,112 @@
+"""
+Key/locator types for Blockstore-based content libraries
+"""
+# Disable warnings about _to_deprecated_string etc. which we don't want to implement:
+# pylint: disable=abstract-method, no-member
+from __future__ import absolute_import, division, print_function, unicode_literals
+
+from opaque_keys import InvalidKeyError
+
+from openedx.core.djangoapps.xblock.learning_context.keys import (
+    check_key_string_field,
+    BlockUsageKeyV2,
+    LearningContextKey,
+)
+
+
+class LibraryLocatorV2(LearningContextKey):
+    """
+    A key that represents a Blockstore-based content library.
+
+    When serialized, these keys look like:
+        lib:hogwarts-p300-potions-exercises
+    """
+    CANONICAL_NAMESPACE = 'lib'
+    KEY_FIELDS = ('slug', )
+    __slots__ = KEY_FIELDS
+    CHECKED_INIT = False
+
+    def __init__(self, slug):
+        """
+        Construct a GlobalUsageLocator
+        """
+        check_key_string_field(slug)
+        super(LibraryLocatorV2, self).__init__(slug=slug)
+
+    def _to_string(self):
+        return self.slug
+
+    @classmethod
+    def _from_string(cls, serialized):
+        return cls(slug=serialized)
+
+    def make_definition_usage(self, definition_key, usage_id=None):
+        """
+        Return a usage key, given the given the specified definition key and
+        usage_id.
+        """
+        return LibraryUsageLocatorV2(library_slug=self.slug, block_type=definition_key.block_type, usage_id=usage_id)
+
+    def for_branch(self, branch):
+        """
+        Compatibility helper.
+        Some code calls .for_branch(None) on course keys. By implementing this,
+        it improves backwards compatibility between library keys and course
+        keys.
+        """
+        if branch is not None:
+            raise ValueError("Cannot call for_branch on a content library key, except for_branch(None).")
+        return self
+
+
+class LibraryUsageLocatorV2(BlockUsageKeyV2):
+    """
+    An XBlock in a Blockstore-based content library.
```

Comment:
Can you add an example of what this key will look like here as well.

Reply:
Example added in a93acae380309589553266becd4889d93178d1bd